vars/withRetryAndTimeout.groovy

```diff
@@ -0,0 +1,31 @@
+
+def call(Map userConfig = [:], Closure body) {
+  def defaultConfig = [
+    retries: 3,
+    bodyTimeout: 10,
+    bodyTimeoutUnit: 'MINUTES',
+  ]
+
+  // Merging the 2 maps - https://blog.mrhaki.com/2010/04/groovy-goodness-adding-maps-to-map_21.html
+  final Map finalConfig = defaultConfig << userConfig
+
+  def counter = 0
+
+  // From https://issues.jenkins.io/browse/JENKINS-51454?focusedCommentId=389893&page=com.atlassian.jira.plugin.system.issuetabpanels%3Acomment-tabpanel#comment-389893
+  retry(finalConfig.retries) {
+    counter += 1
```

Comment:
shouldn't this be incremented at the end? it hasn't retried yet?

Reply:
Depends on if you like 0-indexed or 1-indexed arrays :) The intent is to have the value `1` during the first pass, `2` on the next pass, etc. And the increment at the beginning is to make it easier to "know" but I realize it's not really idiomatic, can lead to errors and the `withEnv` solves this intent.

Reply:
Fixed: let me know if the last change fit what you wanted (If not I need guidance).